src/src/WebServer/DownloadPage.cpp

```diff
@@ -43,10 +64,92 @@ void handle_download()
   {
     str += node_time.getDateTimeString('\0', '\0', '\0');
   }
-  str += F(".dat");
+
+  # if FEATURE_TARSTREAM_SUPPORT
+  bool useTarFile       = false;
+  const int  equalsSign = str.indexOf('=');
+  TarStream *tarStream  = new TarStream(str.substring(equalsSign + 1) + F(".tar"));
+
+  if (fullBackup && (nullptr != tarStream)) {
+    #  if defined(ESP8266)
+
+    fs::Dir dir = ESPEASY_FS.openDir("");
+
+    while (dir.next()) {
+      fs::File f = dir.openFile("r");
+
+      if (f) {
+        tarStream->addFile(f.name(), f.size());
+        f.close();
+      }
+    }
+    #  endif // if defined(ESP8266)
+    #  if defined(ESP32)
+    fs::File root = ESPEASY_FS.open("/");
+    fs::File file = root.openNextFile();
+
+    while (file) {
+      if (!file.isDirectory()) {
+        tarStream->addFile(file.name(), file.size());
+      }
+      file = root.openNextFile();
+    }
+    #  endif // if defined(ESP32)
+  } else {
+    if (nullptr != tarStream) {
+      tarStream->addFile(dataFile.name(), dataFile.size());
+
+      // extcfg<tasknr>.dat files
+      for (uint8_t n = 1; n <= TASKS_MAX; n++) {
```

Comment:
It is working here, but it is rather tricky to iterate over 1 ... TASKS_MAX instead of 0 ... (TASK_MAX-1)
All functions expecting a `taskIndex_t` expect it not to be `TASKS_MAX`.

Reply:
Only need the n +1 value here, that's why I put that in the `for`..., but I ~can~should call `getSettingsFilename(Enum, int)` instead, avoiding the OCD-trigger, it keeps triggering here, every time I look at that line, as well 🤣